pkg/cloudprovider/aws/events/recorder.go

```diff
@@ -0,0 +1,79 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+
+	"github.com/aws/karpenter/pkg/events"
+)
+
+type recorder struct {
+	rec record.EventRecorder
+	events.Recorder
+}
+
+type Recorder interface {
```

Comment:
This events recorder pattern we use across the codebase isn't very DRY. Can we take an AI to investigate how to implement this without so much boilerplate?

Reply:
Sounds good, I can definitely take up an AI to look into this